program/rust/src/rust_oracle.rs

```diff
@@ -22,6 +96,42 @@ pub fn update_version(
     accounts: &Vec<AccountInfo>,
     instruction_data: &[u8],
 ) -> OracleResult {
-    panic!("Need to merge fix to pythd in order to implement this");
+    //read account info
+    let pyth_acc_info_size = size_of::<pc_acc_t>();
+    let mut pyth_acc_info =
+        pc_acc_t::try_from_slice(&accounts[1].try_borrow_data()?[..pyth_acc_info_size])?;
+
+    //check magic number
+    if pyth_acc_info.magic_ != PC_MAGIC {
+        //TODO: once we settle on error stuff fix this
+        return Err(OracleError::IncorrectMagicNumber.into());
+    }
+
+    //update the accounts depending on type
+    match pyth_acc_info.type_ {
+        PC_ACCTYPE_PRICE => upgrade_price_account(program_id, &accounts, instruction_data)?,
+        _ => (),
+    }
+
+    //I do not think that we need to have PC_VERSION dependent logic now, but here seems like a
+    // good place to do such a thing
+
+    //update the account info
+    pyth_acc_info.ver_ = PC_VERSION;
+    //TODO: Should I update the size_ field? I am not sure how it is being used, or why it is
+    // needed. Is it account_size?
+    pyth_acc_info.serialize(&mut &mut accounts[1].data.borrow_mut()[..pyth_acc_info_size])?;
+    Ok(0) //SUCCESS
```

Comment:
This can only return 0 so a ProgramResult would be better suited imo

Reply:
This would break the processor

Reply:
Since, all instructions should have the same return time, otherwise the match statement is going to be wierd